src/builds/node/launch-build.ts

```diff
@@ -0,0 +1,80 @@
+import { IBuildOutputParsedEntry } from './build-parser';
+import { GccBuildParser } from './gcc-build-parser';
+import * as Events from "events"
+import * as child from 'child_process';
+
+
+export interface LaunchBuildinfo {
+    /** the directory where to execute the build */
+    cwd: string
+    /** build executable or script to call */
+    buildCommand: string,
+    /** arguments to pass to build executable/script */
+    buildCommandArgs?: any[], // e.g. "all", clean", etc
+    /** Which parser to use */
+    buildParser: string
+}
+
+export interface ILaunchBuild {
+    launch(launchinfo: LaunchBuildinfo): Promise<string>
+}
+
+export interface ILaunchResult {
+    stdout: NodeJS.ReadableStream,
+    stderr: NodeJS.ReadableStream
+}
+
+export class LaunchBuild extends Events.EventEmitter implements ILaunchBuild {
```

Comment:
LaunchBuild sounds like this should be 2 different things Launch and Build and like Launch should get Build as argument or something or Build just be something more generic as a dependency. Maybe Launch should have LaunchContributions and Build should be one of them.. then you could have like Launch -> execute Build -> execute -> Run etc... (Something to discuss)

I'm also working on making something that exposes a terminal more generic so we should find some synergies there.